app/silhouette/CustomVKProvider.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package silhouette
+
+import com.mohiva.play.silhouette.api.util.HTTPLayer
+import com.mohiva.play.silhouette.impl.providers._
+import models.user.User
+import play.api.libs.json.JsValue
+
+import scala.concurrent.{ExecutionContext, Future}
+import com.mohiva.play.silhouette.impl.providers.oauth2.BaseVKProvider
+import com.mohiva.play.silhouette.impl.providers.oauth2.VKProfileParser
+
+/**
+  * Custom vk profile parser.
+  */
+class CustomVKProfileParser(implicit ec: ExecutionContext) extends CustomSocialProfileParser[JsValue, OAuth2Info] {
```

Comment:
Nit: `CustomVKProfileParser` -> `CustomVkProfileParser`

Reply:
I disagree, VK uses this case on all their sources I see and, more importantly, silhouette follows the same naming scheme, so I don't see any reason not to stay in sync to those

Reply:
But that provider is only used in our app, so shouldn't it follow general convention of the app?

Reply:
What is the general convention of the app?

Reply:
Abbreviations are written in camel case.

Exception is configuration options, but that's the problem of it's own.

Reply:
That is just my opinion, decision is after you

Reply:
Thanks for pointing this out, but I have a strong feeling that this is better, so let's keep this as is